docs/kf_uids/uid_proposal.md

```diff
@@ -0,0 +1,38 @@
+# Proposal
+
+Based on [uid_considerations.md](uid_considerations.md) and [uid_migration.md](uid_migration.md)
+
+## 64 bit keys
+
+### Nodeless (much better than what we have now)
+
+* 31 bit seconds since custom epoch (now until 68 years from now)
+* 33 bit randomized counter (8.5 billion keys for every second)
+* Cross fingers
+
+(Increasing to more bits would make it more better but also more bigger.)
+
+### Nodefull
+
+* 31 bit seconds since custom epoch (now until 68 years from now)
+* 10 bit dataservice instance ID (1024 server instances)
+  * Requires a dedicated ID assignment service (When an autoscale dataservice task spins up, the first thing it does is request a Node ID from a single task node assignment service. The assignment service just increments after each request up to 1024 and then resets.)
+* 23 bit randomized counter (8 million keys in every second)
+
+## Encoding
+
+Studies need to keep old key format for compatibility with Amazon S3 bucket name restrictions, but we can improve the encoding for everything else.
+
+Either base57 or better-than-crockford-base32 (see [uid_considerations.md # Modifying Crockford's Base32 symbol set](uid_considerations.md))
```

Comment:
Decided that, while flawed, Crockford's Base23 symbol set is good enough for our needs and means one less thing changing with the id migration.